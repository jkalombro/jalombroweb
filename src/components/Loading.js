import React from 'react';
import './Loading.scss';
import { SpinnerDotted } from 'spinners-react';

const Loading = () => {
  return (
    <div className="loading-wrapper">
        <SpinnerDotted size={90} thickness={122} speed={136} color="rgba(57, 172, 158, 1)" />
    </div>
  )
}

export default Loading;
