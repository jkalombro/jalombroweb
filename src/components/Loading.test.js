import React from 'react';
import '@testing-library/jest-dom';
import render from '../utils/jest-context-helper';
import Loading from './Loading';

describe("Loading Component", () => {
  it('renders without crashing', async () => {
    const { getByTestId } = render(<Loading />);
    const element = getByTestId("loadingComponent");
    expect(element).toBeTruthy();
  });

  it('matches with snapshot', async () => {
    const { getByTestId } = render(<Loading />);
    const element = getByTestId("loadingComponent");
    expect(element).toMatchSnapshot();
  });
});