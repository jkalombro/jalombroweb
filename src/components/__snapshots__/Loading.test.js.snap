// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component matches with snapshot 1`] = `
<div
  class="loading-wrapper"
  data-testid="loadingComponent"
>
  <svg
    fill="none"
    style="color: rgb(57, 172, 158); overflow: visible; width: 90px;"
    viewBox="0 0 66 66"
  >
    <circle
      cx="33"
      cy="33"
      fill="currentColor"
      r="3.66"
      style="transform: translate(22px, -20px); animation: spinners-react-dotted-shrink 1.4705882352941178s cubic-bezier(0, 0.9, 0, 0.9) 0s infinite;"
    />
    <circle
      cx="33"
      cy="33"
      fill="currentColor"
      r="3.66"
      style="transform: translate(29px, 0px); animation: spinners-react-dotted-shrink 1.4705882352941178s cubic-bezier(0, 0.9, 0, 0.9) 0.07352941176470588s infinite;"
    />
    <circle
      cx="33"
      cy="33"
      fill="currentColor"
      r="3.66"
      style="transform: translate(22px, 20px); animation: spinners-react-dotted-shrink 1.4705882352941178s cubic-bezier(0, 0.9, 0, 0.9) 0.14705882352941177s infinite;"
    />
    <circle
      cx="33"
      cy="33"
      fill="currentColor"
      r="3.66"
      style="transform: translate(0px, 30px); animation: spinners-react-dotted-shrink 1.4705882352941178s cubic-bezier(0, 0.9, 0, 0.9) 0.22058823529411764s infinite;"
    />
    <circle
      cx="33"
      cy="33"
      fill="currentColor"
      r="3.66"
      style="transform: translate(-23px, 20px); animation: spinners-react-dotted-shrink 1.4705882352941178s cubic-bezier(0, 0.9, 0, 0.9) 0.29411764705882354s infinite;"
    />
    <circle
      cx="33"
      cy="33"
      fill="currentColor"
      r="3.66"
      style="transform: translate(-30px, 0px); animation: spinners-react-dotted-shrink 1.4705882352941178s cubic-bezier(0, 0.9, 0, 0.9) 0.36764705882352944s infinite;"
    />
    <circle
      cx="33"
      cy="33"
      fill="currentColor"
      r="3.66"
      style="transform: translate(-23px, -20px); animation: spinners-react-dotted-shrink 1.4705882352941178s cubic-bezier(0, 0.9, 0, 0.9) 0.4411764705882353s infinite;"
    />
    <circle
      cx="33"
      cy="33"
      fill="currentColor"
      r="3.66"
      style="transform: translate(0px, -30px); animation: spinners-react-dotted-shrink 1.4705882352941178s cubic-bezier(0, 0.9, 0, 0.9) 0.5147058823529412s infinite;"
    />
    <circle
      cx="33"
      cy="33"
      fill="currentColor"
      r="7.32"
      style="animation: spinners-react-dotted-center 1.4705882352941178s ease-out infinite; transform-origin: center;"
    />
  </svg>
</div>
`;
